### ISSUE TITLE
TransactionBuilder.CoinSelector can't be modified when calling WalletController.BuildTransaction

### ISSUE THREAD

**Issue by damien-r** [stratisproject/StratisBitcoinFullNode#1534]
The code of StratisBitcoinFullNode is in majority well written to be modular, to allow developers to use it and customize its features without having to touch the original code using dependency injections, interfaces and other mechanisms.

However, I am trying to change the comportment of ICoinSelector when calling WalletController.BuildTransaction, but I can't find a way to do so without having to fork the project. TransactionBuilder have a SetCoinSelector method, but I can't access a pointer to the TransactionBuilder during the processing because it is a local variable and the methods that have access to it are not virtual.

IMHO, there should be a way to inject a custom ICoinSelector in this process, using a Factory injected, by example, or by making WalletTransactionHandler.InitializeTransactionBuilder protected virtual.

**Comment by bokobza** [comment]
Yep, that's a great idea and something we've thought a lot about.
Somehow it didn't make it in the Wallet yet, but we'll try to address this soon.
Thanks @damien-r .

**Comment by rowandh** [comment]
@majikandy @dangershony happy to grab this one if it fits in the next sprint

**Comment by dangershony** [comment]
@damien-r this is a fairly simple change if you want to submit a PR that allows to inject a coin selector and uses the default if none is injected.
@bokobza WDYT
